documentation/src/docs/dev-guide.md

```diff
@@ -0,0 +1,203 @@
+```java
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+```
+Create a datastream that gets input data by connecting to the socket
+```java
+DataStream<String> dataStream = env.socketTextStream(host, port);
+```
+A Pravega Stream may be used as a data sink within a Flink program using an instance of `io.pravega.connectors.flink.FlinkPravegaWriter`. We add an instance of the writer to the dataflow program:
+```java
+FlinkPravegaWriter<String> writer = FlinkPravegaWriter.<String>builder()
+        .withPravegaConfig(pravegaConfig)
+        .forStream(stream)
+        .withSerializationSchema(new SimpleStringSchema())
+        .build();
+dataStream.addSink(writer).name("Pravega Sink");
+```
+Then we execute the job within the Flink environment
+```java
+env.execute("PravegaWriter");
+```
+Executing the above lines should ensure we have created a PravegaWriter job
+
+## Create an application that reads from Pravega
+Creating a Word Count Reader is similar to Word Count Writer
+First create a pravega configuration reading from arguments:
+```java
+ParameterTool params = ParameterTool.fromArgs(args);
+PravegaConfig pravegaConfig = PravegaConfig
+        .fromParams(params)
+        .withDefaultScope("my_scope");
+```
+Initialize the Flink execution environment
+```java
+final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+```
+A Pravega Stream may be used as a data source within a Flink streaming program using an instance of `io.pravega.connectors.flink.FlinkPravegaReader`. The reader reads a given Pravega Stream (or multiple streams) as a DataStream
+```java
+FlinkPravegaReader<String> source = FlinkPravegaReader.<String>builder()
+        .withPravegaConfig(pravegaConfig)
+        .forStream(stream)
+        .withDeserializationSchema(new SimpleStringSchema())
+        .build();
+```
+Then create a datastream count each word over a 10 second time period
+```java
+DataStream<WordCount> dataStream = env.addSource(source).name("Pravega Stream")
+        .flatMap(new Tokenizer()) // The Tokenizer() splits the line into words, and emit streams of "WordCount(word, 1)"
+        .keyBy("word")
+        .timeWindow(Time.seconds(10))
+        .sum("count");
+```
+Create an output sink to print to stdout for verification
+```java
+dataStream.print();
+```
+Then we execute the job within the Flink environment
+```java
+env.execute("PravegaReader");
+```
+
+## Run in flink environment
+First build your application. From Flink's perspective, the connector to Pravega is part of the streaming application (not part of Flink's core runtime), so the connector code must be part of the application's code artifact (JAR file). Typically, a Flink application is bundled as a `fat-jar` (also known as an `uber-jar`) , such that all its dependencies are embedded.
+
+Use gradle or maven to assemble a distribution folder containing the Flink programs as a ready-to-deploy fat-jar
```

Comment:
This requires the assemble plugin in gradle which is not generic, we can just say use the packaged jar, and run 
```
flink run -c <classname> ${your-app}.jar --controller <pravega-controller-uri>
```
We don't need to specify below things like creating socket connection or checking the output, these things will be covered in the dev guide for samples.